ex1/k_namizaki/main1.py

```diff
@@ -0,0 +1,96 @@
+"""
+This code draws a spectrogram from a wavfile and returns it to the original.
+
+FFT a wavfile into a spectrogram.
+IFFT the spectrogram back to the original data.
+"""
+
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.fftpack as fftpack
+import scipy.signal as signal
+import soundfile as sf
+
+
+def main():
+    """
+    Draws a spectrogram from a wavfile and returns it to the original.
+
+    FFT a wavfile into a spectrogram.
+    IFFT the spectrogram back to the original data.
+    """
+    parser = argparse.ArgumentParser(
+        description="wavfileからスペクトログラムを描画し、元に戻す"
+    )
+    parser.add_argument("-file", help="ファイルを入力")
+    args = parser.parse_args()
```

Comment:
argparseを使ってコマンドラインから変数を指定できるように工夫していて素晴らしいですね！
ファイル名だけでなくFFT関連のパラメータもコマンドラインから指定できるようにすると柔軟なプログラムになって良いと思います。
```suggestion
    parser = argparse.ArgumentParser(
        description="wavfileからスペクトログラムを描画し、元に戻す"
    )
    parser.add_argument("-file", help="ファイルを入力")
    parser.add_argument("-n_fft", help="FFTサイズ")
    parser.add_argument("-shift", help="ホップサイズ")
    args = parser.parse_args()
```